### PR TITLE
Deprecate dimensions / metrics APIs on brokers

### PR DESCRIPTION
**Pull request by jihoonson**
Fixes https://github.com/apache/incubator-druid/issues/1810.

### CODE REVIEW

File: docs/content/operations/api-reference.md [hunk 1]
```diff
@@ -412,10 +412,18 @@ druid.query.segmentMetadata.defaultHistory
 
 Returns the dimensions of the datasource.
 
+<div class="note caution">
```
Comment (QiuMM) [inline docs/content/operations/api-reference.md]: Maybe include comments on what to use instead would be better.

Comment (gianm) [reply docs/content/operations/api-reference.md]: segmentMetadata queries or Druid SQL INFORMATION_SCHEMA, probably. And I would include the reason: they provide more comprehensive information, and they work on streaming datasources too, which these APIs do not always.

Comment (jihoonson) [reply docs/content/operations/api-reference.md]: Added.

**Comment by fjy** [comment]
👍

**Pull request merged**